### PR TITLE
remove unnecessary database queries

### PR DESCRIPTION
**Pull request by XzzX**
Improves performance of `remove_jobs`.

### CODE REVIEW

File: pyiron_base/project/generic.py [hunk 1]
```diff
@@ -1700,23 +1700,19 @@ def _remove_jobs_helper(self, recursive=False, progress=True):
         """
         if not isinstance(recursive, bool):
             raise ValueError("recursive must be a boolean")
-        if not self.db.view_mode:
-            job_id_lst = self.get_job_ids(recursive=recursive)
-            if progress and len(job_id_lst) > 0:
-                job_id_lst = tqdm(job_id_lst)
-            for job_id in job_id_lst:
-                if job_id not in self.get_job_ids(recursive=recursive):
-                    continue
```
Comment (jan-janssen) [inline pyiron_base/project/generic.py]: If I understand this pull request correctly, then the primary change is removing this database query, correct? Maybe we should split this pull request in two parts. One containing just the removal of this part and the other the refactoring. Having both mixed makes it a bit hard to read.

Comment (jan-janssen) [reply pyiron_base/project/generic.py]: To clarify these changes a bit more I created two separate pull requests https://github.com/pyiron/pyiron_base/pull/1326 and https://github.com/pyiron/pyiron_base/pull/1325 - these contain the changes to the database queries but not the refactoring done in this pull request.

File: pyiron_base/project/generic.py [hunk 2]
```diff
@@ -1185,27 +1185,27 @@ def remove_job(self, job_specifier, _unprotect=False):
...
+            if _unprotect:
+                job.remove_child()
             else:
-                raise EnvironmentError("copy_to: is not available in Viewermode !")
+                job.remove()
+        except IOError as _:
+            state.logger.debug(
+                "hdf file does not exist. Removal from database will be attempted."
+            )
+            self.db.delete_item(job.id)
```
Comment (XzzX) [inline pyiron_base/project/generic.py:1207]: @jan-janssen You missed this line. It might not be called often but it is another redundant database query.

Comment (jan-janssen) [reply pyiron_base/project/generic.py:1207]: But this is dangerous. For example when you create a job object and remove the HDF5 file, then the loading of the job object with `inspect()` fails and returns an `IOError` still in the except case the job object is not available, so we cannot use `job.id`.

Comment (jan-janssen) [reply pyiron_base/project/generic.py:1207]: Ok, I realised that the `inspect()` function also works when the file is corrupted https://github.com/pyiron/pyiron_base/pull/1327

**Pull request merged**